### PR TITLE
[8.14] API event field updates

### PR DESCRIPTION
**Pull request by jdu2600**
## Change Summary
* Add `Target.process.Ext.protection` field to API events - added in 8.13.0 (and 8.12.3)
* Add `[Target.]process.Ext.created_suspended` field to process and API events - new for 8.14.0
* Add `connect()` parameter fields to API events - new in 8.14.0
* Add TCP/IP metadata source fields - new in 8.14.0

### For mapping changes:
- [x] I ran `make` after making the schema changes, and committed all changes
- [ ] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [ ] If this is a `metadata` change, I also updated both transform destination schemas to match

### CODE REVIEW

File: custom_subsets/elastic_endpoint/api/api.yaml [hunk 1]
```diff
@@ -61,8 +61,10 @@ fields:
           entity_id: {}
           Ext:
             fields:
+              created_suspended: {}
               memory_region:
                 fields: "*"
+              protection: {}
```
Comment (gabriellandau) [inline custom_subsets/elastic_endpoint/api/api.yaml:67]: 💯

File: package/endpoint/data_stream/api/sample_event.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "Target": {
         "process": {
             "Ext": {
+                "created_suspended": true,
```
Comment (gabriellandau) [inline package/endpoint/data_stream/api/sample_event.json:6]:
```suggestion
                "created_suspended": true,
                "protection": "PsProtectedSignerWinTcb",
```

File: custom_schemas/custom_api.yml [hunk 3]
```diff
@@ -426,3 +426,31 @@
...
+      description: >
+        Transport layer protocol. Typically tcp or udp.
+      example: "tcp"
+
+    - name: parameters.ip
+      level: custom
+      type: ip
+      description: >
+        IP address (IPv4 or IPv6).
+      example: "35.238.164.82"
```
Comment (gabriellandau) [inline custom_schemas/custom_api.yml:449]: Is this local IP or remote IP?  Why not use the existing `source.address`/`destination.address` ECS fields?

Comment (jdu2600) [reply custom_schemas/custom_api.yml:449]: It's whatever makes sense for the API in question.  Currently this is always remote, but source/destination would depend on the direction. So destination for `connect()` - but if would be source for `accept()` (not yet implemented).  And potentially neither for some APIs.

Currently we keep all API parameter information together in `process.Ext.parameters`. Scattering them across the various existing top level fields was potentially confusing - though duplication is a possibility.

Comment (gabriellandau) [reply custom_schemas/custom_api.yml:449]: I understand you.  I've held similar viewpoints (and made similar arguments) as we were originally implementing endpoint-package.  I lost 😉 

Say I want to search ES for all network activity from `1.2.3.4`.  I want data from all sources, including Windows Endpoint's WFP filter, macOS Endpoint's network extension, Linux Endpoint's eBPF, PacketBeat, my firewall logs, and Windows ETW sources.  With ECS, I can search for `source.address: 1.2.3.4 OR destination.address: 1.2.3.4` to get it all.  If we spread that data across several different locations depending on the data source, then customers lose that benefit.

Comment (jdu2600) [reply custom_schemas/custom_api.yml]: 👍 7db7644

**Review by gabriellandau — APPROVED** [review]
🚀

**Comment by jdu2600** [comment]
@elastic/security-defend-workflows - any further feedback on this PR?

**Review by pzl — APPROVED** [review]
good to merge

**Comment by jdu2600** [comment]
Thanks @gabriellandau @pzl

**Pull request merged**